diff --git a/BUILD.gn b/BUILD.gn
index c344c9bb1..4132306a5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -23,6 +23,7 @@ config("pdfium_common_config") {
   defines = [
     "PNG_PREFIX",
     "PNG_USE_READ_MACROS",
+    "FPDFSDK_EXPORTS",
   ]
 
   if (!use_system_libopenjpeg2) {
@@ -332,7 +333,7 @@ group("pdfium_public_headers") {
   ]
 }
 
-component("pdfium") {
+shared_library("pdfium") {
   libs = []
   configs += [ ":pdfium_strict_config" ]
   public_configs = [ ":pdfium_public_config" ]
diff --git a/public/fpdfview.h b/public/fpdfview.h
index 779fa5fb9..47f0f233d 100644
--- a/public/fpdfview.h
+++ b/public/fpdfview.h
@@ -196,7 +196,11 @@ typedef int FPDF_OBJECT_TYPE;
 #endif  // defined(FPDF_IMPLEMENTATION)
 #endif  // defined(WIN32)
 #else
+#if defined(WIN32) && defined(FPDFSDK_EXPORTS)
+#define FPDF_EXPORT __declspec(dllexport)
+#else
 #define FPDF_EXPORT
+#endif  // defined(WIN32) && defined(FPDFSDK_EXPORTS)
 #endif  // defined(COMPONENT_BUILD)
 
 #if defined(WIN32) && defined(FPDFSDK_EXPORTS)
