diff --git a/BUILD.gn b/BUILD.gn
index cb1da47df..f8b71effd 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -26,7 +26,9 @@ config("pdfium_common_config") {
   cflags_cc = []
   ldflags = []
   include_dirs = [ "." ]
-  defines = []
+  defines = [
+    "FPDFSDK_EXPORTS",
+  ]
 
   if (!use_system_libopenjpeg2) {
     defines += [ "OPJ_STATIC" ]
@@ -341,7 +343,7 @@ group("pdfium_public_headers") {
   ]
 }
 
-component("pdfium") {
+shared_library("pdfium") {
   output_name = "pdfium"
   libs = []
   configs += [ ":pdfium_strict_config" ]
diff --git a/public/fpdfview.h b/public/fpdfview.h
index 206bfe2d5..48e0fcc20 100644
--- a/public/fpdfview.h
+++ b/public/fpdfview.h
@@ -208,7 +208,11 @@ typedef int FPDF_OBJECT_TYPE;
 #endif  // defined(FPDF_IMPLEMENTATION)
 #endif  // defined(WIN32)
 #else
+#if defined(WIN32) && defined(FPDFSDK_EXPORTS)
+#define FPDF_EXPORT __declspec(dllexport)
+#else
 #define FPDF_EXPORT
+#endif  // defined(WIN32) && defined(FPDFSDK_EXPORTS)
 #endif  // defined(COMPONENT_BUILD)
 
 #if defined(WIN32) && defined(FPDFSDK_EXPORTS)
